web/src/main/java/org/cbioportal/web/PatientController.java

```diff
@@ -14,45 +21,86 @@
+        @RequestParam(required = false) PatientSortBy sortBy,
+        @ApiParam("Direction of the sort")
+        @RequestParam(defaultValue = "ASC") Direction direction) {
 
-        throw new UnsupportedOperationException();
+        if (projection == Projection.META) {
+            HttpHeaders responseHeaders = new HttpHeaders();
+            responseHeaders.add(HeaderKeyConstants.TOTAL_COUNT, patientService.getMetaPatientsInStudy(studyId)
+                .getTotalCount().toString());
+            return new ResponseEntity<>(responseHeaders, HttpStatus.OK);
+        } else {
+            return new ResponseEntity<>(
+                patientService.getAllPatientsInStudy(studyId, projection.name(), pageSize, pageNumber,
+                    sortBy == null ? null : sortBy.name(), direction.name()), HttpStatus.OK);
+        }
     }
 
-    @RequestMapping(value = "/studies/{studyId}/patients/{patientId}", method = RequestMethod.GET)
+    @RequestMapping(value = "/studies/{studyId}/patients/{patientId}", method = RequestMethod.GET,
+        produces = MediaType.APPLICATION_JSON_VALUE)
     @ApiOperation("Get a patient in a study")
-    public ResponseEntity<Patient> getPatientInStudy(@PathVariable String studyId,
-                                                            @PathVariable String patientId) {
+    public ResponseEntity<Patient> getPatientInStudy(
+        @ApiParam(required = true, value = "Study ID e.g. acc_tcga")
+        @PathVariable String studyId,
+        @ApiParam(required = true, value = "Patient ID e.g. TCGA-OR-A5J2")
+        @PathVariable String patientId) throws PatientNotFoundException {
 
-        throw new UnsupportedOperationException();
+        return new ResponseEntity<>(patientService.getPatientInStudy(studyId, patientId), HttpStatus.OK);
     }
 
-    @RequestMapping(value = "/patients/fetch", method = RequestMethod.POST)
+    @RequestMapping(value = "/patients/fetch", method = RequestMethod.POST, consumes = MediaType.APPLICATION_JSON_VALUE,
+        produces = MediaType.APPLICATION_JSON_VALUE)
     @ApiOperation("Fetch patients by ID")
-    public ResponseEntity<List<Patient>> fetchPatients(@RequestParam(defaultValue = "SUMMARY") Projection projection,
-                                                                        @RequestBody List<PatientIdentifier> patientIdentifiers) {
+    public ResponseEntity<List<Patient>> fetchPatients(
+        @ApiParam("Level of detail of the response")
+        @RequestParam(defaultValue = "SUMMARY") Projection projection,
+        @ApiParam(required = true, value = "List of patient identifiers")
+        @RequestBody List<PatientIdentifier> patientIdentifiers) throws PageSizeTooBigException {
 
-        throw new UnsupportedOperationException();
-    }
+        if (patientIdentifiers.size() > PagingConstants.MAX_PAGE_SIZE) {
+            throw new PageSizeTooBigException(patientIdentifiers.size());
+        }
+
+        List<String> studyIds = new ArrayList<>();
```

Comment:
argument processing inside controller : ok
the list construction code is reasonably short ; when the logic gets heavier I would start suggesting a util class